### PR TITLE
LPS-143232 Add alert when the object is seted to Hide Widget

### PR DESCRIPTION
**Pull request by danvb112**
Related Issue: https://issues.liferay.com/browse/LPS-143232

### CODE REVIEW

**Comment by danvb112** [comment]
ci:test:relevant

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/object_entries/view_object_entries.jsp [hunk 1]
```diff
@@ -18,17 +18,30 @@
...
+			namespace="<%= liferayPortletResponse.getNamespace() %>"
+			pageNumber="<%= 1 %>"
+			portletURL="<%= liferayPortletResponse.createRenderURL() %>"
+			style="fluid"
+		/>
+	</c:when>
+	<c:otherwise>
+		<clay:alert
+			displayType="warning"
+			message="This is Object is not available or it was not published."
```
Comment (gabrielwas) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/object_entries/view_object_entries.jsp:43]: We can define this message in the `portal-language-lang` and use it like this example:

[view_flat_organizations.jsp:L144](https://github.com/liferay/liferay-portal/blob/master/modules/apps/users-admin/users-admin-web/src/main/resources/META-INF/resources/view_flat_organizations.jsp#L144-L146)

This way we can make this message localizable

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/object_entries/view_object_entries.jsp [hunk 2]
```diff
@@ -18,17 +18,30 @@
...
-	id="<%= viewObjectEntriesDisplayContext.getClayHeadlessDataSetDisplayId() %>"
-	itemsPerPage="<%= 20 %>"
-	namespace="<%= liferayPortletResponse.getNamespace() %>"
-	pageNumber="<%= 1 %>"
-	portletURL="<%= liferayPortletResponse.createRenderURL() %>"
-	style="fluid"
-/>
\ No newline at end of file
+<c:choose>
+	<c:when test="<%= objectDefinition.isPortlet() %>">
```
Comment (gabrielwas) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/object_entries/view_object_entries.jsp:26]: Just a question: When the object is inactive, does this isPortlet method return `false`? Because I think we can use `isActive()` instead. What do you think?

Comment (danvb112) [reply modules/apps/object/object-web/src/main/resources/META-INF/resources/object_entries/view_object_entries.jsp:26]: Hi @gabrielwas, When the object is inactive or active, `isPortlet()` does not change the value. If we use `isActive`, we might have this situation: Object active, but display set to `Hide Widget` and object will continue to be shown on the page, because the `isPortlet()` is false. So I think we have to use the `isPortlet()`

Comment (danvb112) [reply modules/apps/object/object-web/src/main/resources/META-INF/resources/object_entries/view_object_entries.jsp:26]: But, if we want to bring this alert to inactive functionality, we can add a new condition too, what you think?

Comment (gabrielwas) [reply modules/apps/object/object-web/src/main/resources/META-INF/resources/object_entries/view_object_entries.jsp:26]: Sorry for the misunderstanding, Daniel. I thought that this behavior was for the active/inactive process. You're right. Thanks.

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/object_entries/view_object_entries.jsp (hunk 1 above)
Comment (gabrielwas) [reply modules/apps/object/object-web/src/main/resources/META-INF/resources/object_entries/view_object_entries.jsp:43]: Also, can you rephrase for something like "This object is not available"

Comment (danvb112) [reply modules/apps/object/object-web/src/main/resources/META-INF/resources/object_entries/view_object_entries.jsp:43]: Ok @gabrielwas, I'll do it thanks! 😃

**Comment by danvb112** [comment]
ci:test:relevant

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/object_entries/view_object_entries.jsp (hunk 2 above)
Comment (guilhermedcamacho) [reply modules/apps/object/object-web/src/main/resources/META-INF/resources/object_entries/view_object_entries.jsp:26]: Hey @danvb112 , unfortunately this approach will not work, because the content is being hidden even when the Portlet is accessed from menu. To use this solution, we would need to know if the Portlet is being rendered inside a Display Page.

**Review by guilhermedcamacho — CHANGES_REQUESTED** [review]

**Comment by danvb112** [comment]
ci:test:relevant

**Comment by danvb112** [comment]
ci:test:sf

**Comment by danvb112** [comment]
ci:test:relevant

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/object_entries/view_object_entries.jsp [hunk 3]
```diff
@@ -18,17 +18,32 @@
 
 <%
 ViewObjectEntriesDisplayContext viewObjectEntriesDisplayContext = (ViewObjectEntriesDisplayContext)request.getAttribute(WebKeys.PORTLET_DISPLAY_CONTEXT);
+
+boolean isObjectWidget = viewObjectEntriesDisplayContext.isObjectWidget();
```
Comment (gabrielwas) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/object_entries/view_object_entries.jsp:22]: @danvb112  can we suppress this variable and put `iewObjectEntriesDisplayContext.isObjectWidget()` directly inside the `when` clause? thx

File: modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/entries/display/context/ViewObjectEntriesDisplayContext.java [hunk 4]
```diff
@@ -57,7 +60,17 @@ public ViewObjectEntriesDisplayContext(
 		_portletResourcePermission = portletResourcePermission;
 
 		_apiURL = "/o" + restContextPath;
+
 		_objectRequestHelper = new ObjectRequestHelper(httpServletRequest);
+
```
Comment (gabrielwas) [inline modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/entries/display/context/ViewObjectEntriesDisplayContext.java:65]: I think this code can be moved to the `isObjectWidget` method. Can you do that?

**Review by gabrielwas — CHANGES_REQUESTED** [review]

File: modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/entries/display/context/ViewObjectEntriesDisplayContext.java (hunk 4 above)
Comment (gabrielwas) [reply modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/entries/display/context/ViewObjectEntriesDisplayContext.java:65]: Actually, I think we can move this code to the JSP file

Comment (danvb112) [reply modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/entries/display/context/ViewObjectEntriesDisplayContext.java]: Done!

**Comment by danvb112** [comment]
ci:test:sf

**Comment by danvb112** [comment]
ci:test:relevant

**Comment by gabrielwas** [comment]
LGTM :) Thx @danvb112

**Comment by danvb112** [comment]
ci:reevaluate:1277401_1117

**Comment by danvb112** [comment]
ci:test:relevant
